document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -95,7 +95,7 @@ public static String prepareCast(String field, Type type) {
     } else if (type.equals(Type.BOOLEAN)) {
       return String.format(fmt, field, type);
     } else /* default is string */ {
-      return field;
+      return String.format(fmt, field, type);
```

Comment:
Now, w/o data accessor pattern, we need a cast for string fields too. Pl. see the test case `testFilterWithNestedFiled`

Reply:
Can we also add a code comment mentioning the need for casting strings?

Reply:
Done.